src/apb/engine.py

```diff
@@ -111,47 +140,74 @@ def make_friendly(blob):
-    actions_path = os.path.join(project_path, ACTIONS_DIR)
-    src_file = 'shipit-%s.yml' % provider
-    provision_src_path = os.path.join(project_path, 'ansible', src_file)
-    provision_dest_path = os.path.join(actions_path, 'provision.yaml')
-
-    if not os.path.exists(actions_path):
-        os.makedirs(actions_path)
-
-    if ansible_dir_exists:
-        # Only write over provision.yml if the file doesn't already exist
-        if not os.path.exists(provision_dest_path):
-            copyfile(provision_src_path, provision_dest_path)
-    else:
-        print('NOTE: No ansible dir found at project root.' +
-              'Assuming manual authoring.')
-
-
-def init_dockerfile(spec_path, dockerfile_path, ansible_dir_exists):
+def update_dockerfile(spec_path, dockerfile_path):
     # TODO: Defensively confirm the strings are encoded
     # the way the code expects
     blob = base64.b64encode(load_spec_str(spec_path))
     dockerfile_out = insert_encoded_spec(
-        load_dockerfile(ansible_dir_exists), make_friendly(blob)
+        load_dockerfile(dockerfile_path), make_friendly(blob)
     )
 
-    write_dockerfile(dockerfile_out, dockerfile_path)
+    write_dockerfile(dockerfile_out, dockerfile_path, False)
     print('Finished writing dockerfile.')
 
 
+def cmdrun_init(**kwargs):
+    print("Initializing current directory for an APB")
+    current_path = kwargs['base_path']
+    apb_name = kwargs['name']
+    project = os.path.join(current_path, apb_name)
+
+    if os.path.exists(project):
+        if not kwargs['force']:
+            raise Exception('ERROR: Project directory: [%s] found and force option not specified' % project)
+        shutil.rmtree(project)
+
+    os.mkdir(project)
+
+    spec_path = os.path.join(project, SPEC_FILE)
+    playbooks_path = os.path.join(project, PLAYBOOKS_DIR)
+    roles_path = os.path.join(project, ROLES_DIR)
+    dockerfile_path = os.path.join(os.path.join(project, DOCKERFILE))
+
+    if os.path.exists(spec_path) and not kwargs['force']:
+        raise Exception('ERROR: Spec file: [%s] found and force option not specified' % spec_path)
+
+    specfile_out = load_example_specfile()
+    write_specfile(specfile_out, spec_path, kwargs['force'])
+
+    dockerfile_out = load_dockerfile(EX_DOCKERFILE_PATH)
+    write_dockerfile(dockerfile_out, dockerfile_path, kwargs['force'])
+
+    if os.path.exists(playbooks_path):
+        if not kwargs['force']:
+            raise Exception('ERROR: Playbooks dir: [%s] was found and force option not specified. Forcing a reinit will recreate the directory.' % playbooks_path)
+        shutil.rmtree(playbooks_path)
+
+    os.mkdir(playbooks_path)
+
+    if os.path.exists(roles_path):
+        if not kwargs['force']:
+            raise Exception('ERROR: Roles dir: [%s] was found and force option not specified. Forcing a reinit will recreate the directory.' % role_path)
+        shutil.rmtree(roles_path)
+
+    os.mkdir(roles_path)
```

Comment:
I would move the `if os.path.exists` / `os.mkdir` up and do that all before you read in the files.

```
    if os.path.exists(project):
        if not kwargs['force']:
            raise Exception('ERROR: Project directory: [%s] found and force option not specified' % project)
        shutil.rmtree(project)

    os.mkdir(project)

    spec_path = os.path.join(project, SPEC_FILE)
    playbooks_path = os.path.join(project, PLAYBOOKS_DIR)
    roles_path = os.path.join(project, ROLES_DIR)
    dockerfile_path = os.path.join(os.path.join(project, DOCKERFILE))

    if os.path.exists(spec_path) and not kwargs['force']:
        raise Exception('ERROR: Spec file: [%s] found and force option not specified' % spec_path)

    if os.path.exists(playbooks_path):
        if not kwargs['force']:
            raise Exception('ERROR: Playbooks dir: [%s] was found and force option not specified. Forcing a reinit will recreate the directory.' % playbooks_path)
        shutil.rmtree(playbooks_path)

    os.mkdir(playbooks_path)

    if os.path.exists(roles_path):
        if not kwargs['force']:
            raise Exception('ERROR: Roles dir: [%s] was found and force option not specified. Forcing a reinit will recreate the directory.' % role_path)
        shutil.rmtree(roles_path)

    os.mkdir(roles_path)

    # NOW do the loading and writing of the files
    specfile_out = load_example_specfile()
    write_specfile(specfile_out, spec_path, kwargs['force'])

    dockerfile_out = load_dockerfile(EX_DOCKERFILE_PATH)
    write_dockerfile(dockerfile_out, dockerfile_path, kwargs['force'])
```

That way you get all of the directories out of the way before you write any files down.